exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusUnitsHelper.java

```diff
@@ -0,0 +1,166 @@
+                {"J", "joules"},
+                {"W", "watts"},
+                {"g", "grams"},
+                // Misc
+                {"Cel", "celsius"},
+                {"Hz", "hertz"},
+                {"1", ""},
+                {"%", "percent"},
+                {"$", "dollars"}
+              })
+          .collect(
+              Collectors.collectingAndThen(
+                  Collectors.toMap(
+                      keyValuePair -> keyValuePair[0], keyValuePair -> keyValuePair[1]),
+                  Collections::unmodifiableMap));
+
+  // The map that translates the "per" unit
+  // Example: s => per second (singular)
+  private static final Map<String, String> PROMETHEUS_PER_UNIT_MAP =
+      Stream.of(
+              new String[][] {
+                {"s", "second"},
+                {"m", "minute"},
+                {"h", "hour"},
+                {"d", "day"},
+                {"w", "week"},
+                {"mo", "month"},
+                {"y", "year"}
+              })
+          .collect(
+              Collectors.collectingAndThen(
+                  Collectors.toMap(
+                      keyValuePair -> keyValuePair[0], keyValuePair -> keyValuePair[1]),
+                  Collections::unmodifiableMap));
+
+  private PrometheusUnitsHelper() {
+    // Prevent object creation for utility classes
+  }
+
+  /**
+   * A utility function that returns the equivalent Prometheus name for the provided OTLP metric
+   * unit.
+   *
+   * @param rawMetricUnitName The raw metric unit for which Prometheus metric unit needs to be
+   *     computed.
+   * @param metricType The {@link PrometheusType} of the metric whose unit is to be converted.
+   * @return the computed Prometheus metric unit equivalent of the OTLP metric unit.
+   */
+  public static String getEquivalentPrometheusUnit(
+      String rawMetricUnitName, PrometheusType metricType) {
+    if (StringUtils.isNullOrEmpty(rawMetricUnitName)) {
+      return rawMetricUnitName;
+    }
+
+    // special case
+    if (rawMetricUnitName.equals("1") && metricType == PrometheusType.GAUGE) {
+      return "ratio";
+    }
+
+    String convertedMetricUnitName = rawMetricUnitName;
+    // Drop units specified between curly braces
+    convertedMetricUnitName = removeUnitPortionInBrackets(convertedMetricUnitName);
+
+    // Handling for the "per" unit(s), e.g. foo/bar -> foo_per_bar
+    if (convertedMetricUnitName.contains("/")) {
+      convertedMetricUnitName = convertRateExpressedToPrometheusUnit(convertedMetricUnitName);
+    }
+
+    // Converting abbreviated unit names to full names
+    return cleanUpString(
+        PROMETHEUS_UNIT_MAP.getOrDefault(convertedMetricUnitName, convertedMetricUnitName));
+  }
+
+  private static String convertRateExpressedToPrometheusUnit(String rateExpressedUnit) {
+    String[] rateEntities = rateExpressedUnit.split("/", 2);
+    // Only convert rate expressed units if it's a valid expression
+    if (rateEntities[1].equals("")) {
```

Comment:
Document that you require a "/" to exist in the string before calling the function.

As an alternative, you could move the "/" check here as an early exit that just returns the passed in string.

Reply:
Right, this sounds good - this way method will be self-sufficient even though its a private one.